src/wallet/tx_builder.rs

```diff
@@ -646,14 +649,22 @@ impl<'a, B, D: BatchDatabase> TxBuilder<'a, B, D, DefaultCoinSelectionAlgorithm,
 }
 
 /// Ordering of the transaction's inputs and outputs
-#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Hash, Clone, Copy)]
+#[derive(Clone)]
 pub enum TxOrdering {
     /// Randomized (default)
     Shuffle,
     /// Unchanged
     Untouched,
     /// BIP69 / Lexicographic
+    #[deprecated = "BIP69 does not improve privacy as was the intention of the BIP"]
     Bip69Lexicographic,
+    /// Provide custom comparison functions for sorting
+    Custom {
```

Comment:
I think we may have to add some sort of "state" here, something that the two functions could look at to know what to do essentially. It doesn't have to be mutable, that would mess a lot of this up.

If we don't wanna add generic types we could maybe add a `Box<dyn std::any::Any>`, which you can then downcast to whatever you need in the functions. Obviously you would also have to add that as an extra argument to your functions.

Reply:
CC @LLFourn who proposed sorting with a shared secret. I think you would not be able to do that, unless you introduce this state.

I'm not sure what happens if you try borrowing from local variables in the two `Fn`s, I'm guessing you are not allowed to do that because of lifetime issues. Maybe we should test this first, actually.

Reply:
@FadedCoder can you write a doc comment or a test which maybe demos how to use a secret to hash the input/output to produce the key? 

Given the awkward `Arc` thing it might be better to have a convenience method like `set_custom_ordering(i_f: impl Fn(&Txin...), o_f: impl Fn(&TxOut,..))` which would wrap the `Arcs` around them.

> I think we may have to add some sort of "state" here, something that the two functions could look at to know what to do essentially. It doesn't have to be mutable, that would mess a lot of this up.

Just `move` in what you need?

Reply:
> Just move in what you need?

For some reason I always instinctively think that `move` will only work with `FnOnce` because I feel like the value would be consumed at the first iteration.

So yeah, `move` probably works, but as you said it would be nice to have an example and more tests to verify this.

Reply:
Yeah sure, I'll add some tests or examples soon!